apps/system/js/notifications.js

```diff
@@ -403,20 +403,24 @@ var NotificationScreen = {
      */
     var dir = (detail.bidi === 'auto' || typeof detail.bidi === 'undefined') ?
       document.documentElement.dir : detail.bidi;
-
+    // As per the UX,in case for sms notification ,the notification title
+    // will be the count along with the 'new message' text,but currently in
+    // 'detail' object, name of the sender is coming. in other cases title text
+    // in 'detail' object is appropriate.
+    var cIndex = this.unreadNotifications.has(manifestURL);
+    if (!cIndex) {
     // We need to animate the ambient indicator when the toast
     // timesout, so we skip updating it here, by passing a skip bool
-    this.addUnreadNotification(detail.id, true);
-
+    this.addUnreadNotification(manifestURL, true);
```

Comment:
The source code with wrong indentation can't be integrated.

When I mentioned separation on 2 commits, the idea was:

1 - Initial state without any your changes
```javascript
code_line: 1;
code_line: 2;
```
2 - Your comment number 1 (wrap this code with ``{}`` and add indentation)
```javascript
if (magic) {
  code_line: 1;
  code_line: 2;
} else {
...
}
```
3 - Your comment number 2 (just to highlight changes in indented code)
```javascript
if (magic) {
  code_line: 1;
  your_new: code-line;  // made required code changes add/modify/remove anything you need
  code_line: 2;
} else {
...
}
```
As a result of such sequence of 2 commits you will have: 1) all required changes; 2) review of them will be as simple as possible.
In your case you have second commit without must-have changes, so to be able to integrate such PR you anyway will need to create one more commit containing everything what shall be integrated.